From oliver@neukum.name  Fri Nov 24 00:55:17 2006
Date: Thu, 23 Nov 2006 12:45:31 +0100 (MET)
From: Oliver Neukum <oliver@neukum.name>
To: greg@kroah.com, linux-usb-devel@lists.sourceforge.net, dbrownell@users.sourceforge.net, David Hollis <dhollis@davehollis.com>
Subject: USB: endianness fix for asix.c
MIME-Version: 1.0
Content-Type: text/plain;  charset="us-ascii"
Content-Disposition: inline
Message-Id: <200611231237.32250.oliver@neukum.name>

the latest update for asix.c reverted some endianness fixes. This
reinstates them.

Signed-off-by: Oliver Neukum <oliver@neukum.name>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/net/asix.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- gregkh-2.6.orig/drivers/usb/net/asix.c
+++ gregkh-2.6/drivers/usb/net/asix.c
@@ -249,9 +249,9 @@ asix_write_cmd_async(struct usbnet *dev,
 
 	req->bRequestType = USB_DIR_OUT | USB_TYPE_VENDOR | USB_RECIP_DEVICE;
 	req->bRequest = cmd;
-	req->wValue = value;
-	req->wIndex = index;
-	req->wLength = size;
+	req->wValue = cpu_to_le16(value);
+	req->wIndex = cpu_to_le16(index);
+	req->wLength = cpu_to_le16(size);
 
 	usb_fill_control_urb(urb, dev->udev,
 			     usb_sndctrlpipe(dev->udev, 0),
